neuralprophet/time_dataset.py

```diff
@@ -176,7 +180,27 @@ def filter_samples_after_init(
         """
         if prediction_frequency is None or prediction_frequency == 1:
             return
-        self.samples = self.samples[::prediction_frequency]
+        # Only the first target timestamp is of interest for filtering
+        timestamps = pd.to_datetime([sample["timestamps"][0] for sample in self.samples])
+        for key, value in prediction_frequency.items():
+            if key == "daily":
+                mask = timestamps.hour == value + 1  # + 1 because prediction starts one step after origin
+            elif key == "weekly":
+                mask = timestamps.dayofweek == value + 1
+            elif key == "monthly":
+                mask = timestamps.day == value + 1
+            elif key == "yearly":
+                mask = timestamps.month == value + 1
+            elif key == "hourly":
+                mask = timestamps.minute == value + 1
+            else:
+                raise ValueError(f"Invalid prediction frequency: {key}")
+        self.samples = [self.samples[i] for i in range(len(self.samples)) if mask[i]]
+
+        # Exact timestamps are not needed anymore
+        self.inputs.pop("timestamps")
+        for sample in self.samples:
+            sample.pop("timestamps")
```

Comment:
Filtering the samples according to the prediction frequency. See description of the PR to see how the dictionary prediction_frequency works. 
To achieve this, the real timestamps are passed to the samples, but then dropped after filtering is done as we only need them to filter the samples and for nothing else. This keeps the samples lean